Restructure readme and add stage libs to distribution

laserval opened the pull request:
- Readme is better structured with the getting started at the top, instead of compilation instructions.
- Most stage libraries are now found in `distribution/lib` after building the parent pom.

Commit:
Adds lib directory to distribution, containing all built stage libraries

Commit:
Restructures readme, making getting started clearer

Commit:
Reformat

remen commented on distribution/pom.xml:
```diff
@@ -41,8 +41,16 @@
 							<tasks>
 								<copy file="../core/target/hydra-core-jar-with-dependencies.jar" tofile="bin/hydra-core.jar" />
-								<copy file="../core/target/resource.properties" tofile="bin/resource.properties" />
+								<copy file="../core/target/resource.properties" tofile="bin/hydra-core.properties" />
```
When adding stuff here, it's a good idea to add the projects as dependencies as well so that the reactor build is guaranteed to be in the correct order.

remen commented on admin-service/pom.xml:
```diff
@@ -13,22 +13,40 @@
+                        <phase>compile</phase>
+                        <configuration>
+                            <tasks>
+                                <copy file="src/main/resources/admin-service.properties" tofile="target/admin-service.properties" />
```
This double copying surprised me, but I'm guessing there's a good reason for it.

laserval replied:
I was mostly following the "pattern" from Hydra Core. If you're only building this project, I think it makes sense to add the config file next to the binary.

It is a bit unnecessary, though. Do you think we should remove it from both Core and the admin service?

remen commented:
Looking good. Add dependencies to ensure correct reactor ordering and merge.

Commit:
Adds stage libraries as dependencies to distribution to ensure build order. Adds remaining stage libraries to distribution as well.

laserval commented:
I updated the distribution pom with the dependencies, and added the two other libraries I hadn't added (groovy-runner and debugging). Might as well have all stages in there! Had to add jar-with-dependencies packaging to groovy-runner, which goes to show how much that stage has been used...
